From e102282467ca546824ab2c84f11f6e0edd62d58a Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Sun, 8 Dec 2024 12:35:20 +0300
Subject: [PATCH 6/7] base: providing a choice of additional vndk to build
 image with ext4

---
 base.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/base.mk b/base.mk
index ac1e297..553d5a5 100644
--- a/base.mk
+++ b/base.mk
@@ -253,5 +253,3 @@ PRODUCT_PACKAGES += \
 $(call inherit-product, $(SRC_TARGET_DIR)/product/window_extensions.mk)
 PRODUCT_PRODUCT_PROPERTIES += \
     persist.settings.large_screen_opt.enabled=true
-
-PRODUCT_EXTRA_VNDK_VERSIONS += 28 29
-- 
2.34.1

